src/Nethermind/Nethermind.Evm/Precompiles/Bls/G1MulPrecompile.cs

```diff
@@ -24,20 +25,48 @@ private G1MulPrecompile()
 
     public long BaseGasCost(IReleaseSpec releaseSpec) => 12000L;
 
-    public long DataGasCost(in ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec) => 0L;
+    public long DataGasCost(ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec) => 0L;
 
-    public (ReadOnlyMemory<byte>, bool) Run(in ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec)
+    public (ReadOnlyMemory<byte>, bool) Run(ReadOnlyMemory<byte> inputData, IReleaseSpec releaseSpec)
     {
-        const int expectedInputLength = 2 * BlsParams.LenFp + BlsParams.LenFr;
+        const int expectedInputLength = BlsParams.LenG1 + BlsParams.LenFr;
         if (inputData.Length != expectedInputLength)
         {
             return IPrecompile.Failure;
         }
 
-        Span<byte> output = stackalloc byte[2 * BlsParams.LenFp];
-        bool success = SubgroupChecks.G1IsInSubGroup(inputData.Span[..(2 * BlsParams.LenFp)])
-            && Pairings.BlsG1Mul(inputData.Span, output);
+        try
+        {
+            G1 x = new G1(stackalloc long[G1.Sz]);
+            x.DecodeRaw(inputData[..BlsParams.LenG1].Span);
+
+            if (x.IsInf())
+            {
+                return (Enumerable.Repeat<byte>(0, 128).ToArray(), true);
+            }
+
+            if (!x.InGroup())
+            {
+                return IPrecompile.Failure;
+            }
+
+            if (!inputData.Span[BlsParams.LenG1..].ContainsAnyExcept((byte)0))
+            {
+                return (Enumerable.Repeat<byte>(0, 128).ToArray(), true);
+            }
+
+            Span<byte> scalar = stackalloc byte[32];
+            for (int i = 0; i < 32; i++)
+            {
+                scalar[32 - i - 1] = inputData.Span[BlsParams.LenG1 + i];
```

Comment:
Can we do Copy?

Reply:
Yes we could copy and then reverse the span (effectively two copies), but this way we do the copy and reverse in one

Reply:
No, you are doing 32x byte operations instead of doing 2x 32byte operations, which will be faster instead of SIMD